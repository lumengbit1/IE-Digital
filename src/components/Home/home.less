.homePage {
    display: flex;
    justify-content: center;
    align-items: flex-start;
    height: 100vh;
    .block {
        display: flex;
        flex-direction: column;
        justify-content: flex-start;
        align-items: center;
        border: 1px dashed #000;
        border-radius: 10px;
        padding: 20px;
        background-color: #fafbfa;
        min-width: 300px;
        min-height: 300px;
        margin-top: 50px;
        .input {
            overflow: visible;
            width: 100%;
            height: 100%;
        }
        .button {
            display: flex;
            button {
                width: 100px;
                margin: 20px;
            }
        }
    }
}
