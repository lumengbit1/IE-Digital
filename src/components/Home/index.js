import React, { useState } from 'react';
import CSSModules from 'react-css-modules';
import { inject } from 'mobx-react';
import { observer } from 'mobx-react-lite';
import styles from './home.less';

function Home({ rootStore }) {
    const updateProperty = e => {
        rootStore.formStore.updateProperty(e.target.value);
        setValue(e.target.value);
    };

    const btnClick = () => {
        setValue('');
        rootStore.formStore.clearOutput();
    };
    const [value, setValue] = useState();
    return (
        <div styleName="homePage">
            <div styleName="block">
                <textarea
                    rows="10"
                    cols="30"
                    data-testid="test_textarea"
                    name="position"
                    styleName="input"
                    onChange={updateProperty}
                    value={value}
                />
                <p>{rootStore.formStore.output}</p>
                <div styleName="button">
                    <button onClick={() => rootStore.formStore.calculate()}>Start</button>
                    <button onClick={() => btnClick()}>Reset</button>
                </div>
            </div>
        </div>
    );
}
export default inject('rootStore')(observer(CSSModules(Home, styles)));
