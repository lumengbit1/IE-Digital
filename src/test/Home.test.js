import React from 'react';
import RootStore from '../stores/RootStore';
import { Provider } from 'mobx-react';
import Home from '../components/Home';
import { render, fireEvent, cleanup } from 'react-testing-library';
import sinon from 'sinon';

const rootStore = new RootStore();
afterEach(cleanup);
describe('Render Test', function () {
    it('case: expect rendering correct', () => {
        const { container } = render(
            <Provider rootStore={rootStore}>
                <Home />
            </Provider>
        );
        expect(container).toMatchSnapshot();
    });
});

describe('Button Click Test', function () {
    it('case: expect calculate be called when Start button be clicked', () => {
        const calculate = sinon.stub(rootStore.formStore, 'calculate');
        const { getByText } = render(
            <Provider rootStore={rootStore}>
                <Home />
            </Provider>
        );
        fireEvent.click(getByText('Start'));
        expect(calculate.called).toEqual(true);
    });

    it('case: expect clearOutput be called when Start button be clicked', () => {
        const clearOutput = sinon.stub(rootStore.formStore, 'clearOutput');
        const { getByText } = render(
            <Provider rootStore={rootStore}>
                <Home />
            </Provider>
        );
        fireEvent.click(getByText('Reset'));
        expect(clearOutput.called).toEqual(true);
    });
    it('case: expect updateProperty be called when input onChange event be triggered', () => {
        const updateProperty = sinon.stub(rootStore.formStore, 'updateProperty');
        const { getByTestId } = render(
            <Provider rootStore={rootStore}>
                <Home />
            </Provider>
        );
        fireEvent.change(getByTestId('test_textarea'), { target: { value: 'TEST VALUE' } });
        expect(updateProperty.called).toEqual(true);
    });
});
