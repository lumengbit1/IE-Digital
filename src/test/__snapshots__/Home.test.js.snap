// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Test case: expect rendering correct 1`] = `
<div>
  <div
    class="homePage"
  >
    <div
      class="block"
    >
      <textarea
        class="input"
        cols="30"
        data-testid="test_textarea"
        name="position"
        rows="10"
      />
      <p />
      <div
        class="button"
      >
        <button>
          Start
        </button>
        <button>
          Reset
        </button>
      </div>
    </div>
  </div>
</div>
`;
